downstairs/src/lib.rs

```diff
@@ -1000,33 +1006,30 @@ where
              * dependencies are met.
              */
             let job_id = ads
-                .lock()
+                .read()
                 .await
                 .in_progress(upstairs_connection, *new_id)
                 .await?;
 
             if let Some(job_id) = job_id {
-                let m = ads
-                    .lock()
-                    .await
-                    .do_work(upstairs_connection, job_id)
-                    .await?;
-
-                if let Some(m) = m {
-                    ads.lock()
-                        .await
-                        .complete_work_stat(upstairs_connection, &m, job_id)
-                        .await?;
-                    // Notify the upstairs before completing work
-                    let mut fw = fw.lock().await;
-                    fw.send(&m).await?;
-                    drop(fw);
-
-                    ads.lock()
-                        .await
-                        .complete_work(upstairs_connection, job_id, m)
-                        .await?;
-                }
+                let ads = ads.clone();
+                let fw = fw.clone();
+                let tx = job_channel_tx.clone();
+                let span = span!(tracing::Level::INFO, "job", job_id);
+
+                tokio::spawn(
+                    async move {
+                        let _res = do_work_subtask(
```

Comment:
It does silently get ignored, yeah.

It might be a good idea to introduce a `Err<CrucibleError>` channel that all these tasks and subtasks can push into, where `resp_loop` can rx out of and bubble up errors that way? What do you think?

Reply:
I think we need to think about it a bit more :)

I do think, for the most part, we see if a task has died and the error handling cleans up.  It's just a matter of leaving breadcrumbs behind so we know what caused the problem.  If we switch to having a common error channel, I worry that the error handling we have now won't work the same.